rpcs3/Emu/RSX/VK/vkutils/sampler.cpp

```diff
@@ -4,9 +4,46 @@
 
 namespace vk
 {
+	extern VkBorderColor get_border_color(u32);
+
+	static VkBorderColor get_closest_border_color_enum(const color4f& color4)
+	{
+		if ((color4.r + color4.g + color4.b) > 1.35f)
+		{
+			//If color elements are brighter than roughly 0.5 average, use white border
+			return VK_BORDER_COLOR_FLOAT_OPAQUE_WHITE;
+		}
+
+		if (color4.a > 0.5f)
+			return VK_BORDER_COLOR_FLOAT_OPAQUE_BLACK;
+
+		return VK_BORDER_COLOR_FLOAT_TRANSPARENT_BLACK;
+	}
+
+	border_color_t::border_color_t(u32 encoded_color)
+		: storage_key(0)
+	{
+		value = vk::get_border_color(encoded_color);
+
+		if (value != VK_BORDER_COLOR_FLOAT_CUSTOM_EXT)
+		{
+			// Nothing to do
+			return;
```

Comment:
color_value is uninitialized in this case (unless it has a proper default constructor)

Reply:
It's a color4f object which default initializes to black.